File: src/main/java/be/zvz/kotlininside/http/HttpException.java
package be.zvz.kotlininside.http;

import org.jetbrains.annotations.Nullable;

public class HttpException extends RuntimeException {
    private final int statusCode;
    @Nullable
    private Throwable cause;

    public HttpException(int statusCode, @Nullable Throwable cause) {
        super(cause);
        this.statusCode = statusCode;
    }

    public HttpException(@Nullable Throwable cause) {
        this(520, cause);
        this.cause = cause;
    }

    public int getStatusCode() {
        return statusCode;
    }

    @Nullable
    public Throwable getCause() {
        return this.cause;
    }
}

File: src/main/kotlin/be/zvz/kotlininside/http/HttpException.kt
package be.zvz.kotlininside.http

class HttpException : RuntimeException {
    val statusCode: Int

    constructor(
        statusCode: Int,
        errorMessage: String?
    ) : super("Error Code : $statusCode, Error Message : $errorMessage") {
        this.statusCode = statusCode
    }

    constructor(cause: Throwable?) : super(cause) {
        statusCode = 520
    }
}
